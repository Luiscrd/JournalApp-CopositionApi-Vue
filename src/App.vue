<template>
  <Header />
  <router-view></router-view>
  <Footer />
</template>

<script>
import { defineAsyncComponent } from 'vue'
import useAuth from '@/modules/auth/composables/useAuth'

export default {
  name: 'App',
  components: {
    Header: defineAsyncComponent( () => import('@/modules/shared/components/Header.vue')),
    Footer: defineAsyncComponent( () => import('@/modules/shared/components/Footer.vue')),
  },
  setup() {
    const { auhtStatus, checkAuthentication } = useAuth()
    checkAuthentication()
    return {
      auhtStatus
    }
  }
}
</script>

<style>
html, body{
  margin: 0;
  padding: 0;
}

body{
  font-family: Arial, Helvetica, sans-serif;
  position: relative;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
}

#app{
  position: relative;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
}
</style>
