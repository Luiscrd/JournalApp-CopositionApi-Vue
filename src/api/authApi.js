import axios from 'axios'

const authApi = axios.create({
    baseURL: 'https://identitytoolkit.googleapis.com/v1/accounts',
    params: {
        key: 'AIzaSyCRSBstjYJSunzlnMVVbJjAa2xqZcwOQLQ'
    }
})

export default authApi