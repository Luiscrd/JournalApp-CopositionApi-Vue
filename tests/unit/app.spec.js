import { shallowMount } from '@vue/test-utils'
import App from '@/App.vue'
import authApi from '@/api/authApi'

describe('App Component', () => {

    beforeAll(async () => {

        const { data } = await authApi.post(':signInWithPassword', {
            email: 'luiscrua@gmail.com',
            password: '123456',
            returnSecureToken: true
        })

        localStorage.setItem('idToken', data.isToken)

    })

    let wrapper;

    beforeEach(() => {

        wrapper = shallowMount(App)

    });

    test('Debe de hacer mach con el snapshot', () => {

        expect(wrapper.html()).toMatchSnapshot()

    })

})