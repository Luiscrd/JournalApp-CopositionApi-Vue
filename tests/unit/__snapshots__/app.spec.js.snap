// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Debe de hacer mach con el snapshot 1`] = `
<header-stub></header-stub>
<router-view></router-view>
<footer-stub></footer-stub>
`;
